File: java/src/com/esielkar/builder/BuilderExample.java
package com.esielkar.builder;

public class BuilderExample {
    public static void main(String[] args) {
        ConcreteBuilder builder = new ConcreteBuilder();

        Director director = new Director(builder);
        director.make(1);
        System.out.println(builder.build());

        builder.reset();
        director.changeBuilder(builder);
        director.make(2);
        System.out.println(builder.build());

        builder.reset();
        builder.buildStepB();
        builder.buildStepC();
        System.out.println(builder.build());

    }

}

File: dart/design_patterns/bin/builder/builder_example.dart
import 'builder.dart';
import 'director.dart';

void main(List<String> args) {
  final builder = ProductBuilder();

  builder.reset();
  builder.buildStepB();
  builder.buildStepC();
  print(builder.build());

  builder.reset();
  final director = Director(builder);
  director.make(1);
  print(builder.build());
}
